src/nft/components/collection/CollectionNfts.tsx

```diff
@@ -1,31 +1,178 @@
 import clsx from 'clsx'
 import useDebounce from 'hooks/useDebounce'
 import { AnimatedBox, Box } from 'nft/components/Box'
-import { FilterButton } from 'nft/components/collection'
+import { CollectionSearch, FilterButton } from 'nft/components/collection'
 import { CollectionAsset } from 'nft/components/collection/CollectionAsset'
 import * as styles from 'nft/components/collection/CollectionNfts.css'
-import { Row } from 'nft/components/Flex'
-import { Center } from 'nft/components/Flex'
+import { SortDropdown } from 'nft/components/common/SortDropdown'
+import { Center, Row } from 'nft/components/Flex'
+import { NonRarityIcon, RarityIcon } from 'nft/components/icons'
 import { bodySmall, buttonTextMedium, header2 } from 'nft/css/common.css'
-import { useCollectionFilters, useFiltersExpanded, useIsMobile } from 'nft/hooks'
+import { vars } from 'nft/css/sprinkles.css'
+import {
+  CollectionFilters,
+  initialCollectionFilterState,
+  SortBy,
+  SortByPointers,
+  Trait,
+  useCollectionFilters,
+  useFiltersExpanded,
+  useIsMobile,
+} from 'nft/hooks'
 import { AssetsFetcher } from 'nft/queries'
-import { UniformHeight, UniformHeights } from 'nft/types'
-import { useEffect, useMemo, useState } from 'react'
+import { DropDownOption, GenieAsset, GenieCollection, UniformHeight, UniformHeights } from 'nft/types'
+import qs from 'query-string'
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
 import InfiniteScroll from 'react-infinite-scroll-component'
 import { useInfiniteQuery } from 'react-query'
+import { useLocation } from 'react-router-dom'
+
+const rarityStatusCache = new Map<string, boolean>()
+function getRarityStatus(id: string, assets?: (GenieAsset | undefined)[]) {
```

Comment:
`rarityStatusCache` should really be declared inside the function. Also we can put `getRarityStatus` inside of `asset.ts` util file